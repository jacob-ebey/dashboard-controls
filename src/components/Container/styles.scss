.container {
  width: 95%;
  height: 100%;
  max-width: 980px;
  margin: 0 auto;
  padding: 0;

  @media (min-width: 1045px) {
    width: 100%;
  }
}
