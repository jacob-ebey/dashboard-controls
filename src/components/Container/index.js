import React from 'react';

import styles from './styles.scss';

export default class Component extends React.Component {
  render() {
    const { className, children, ...rest } = this.props;
    return (
      <div
        className={`${styles.container} ${className || ''}`}
        {...rest}
      >
        {children}
      </div>
    );
  }
}
