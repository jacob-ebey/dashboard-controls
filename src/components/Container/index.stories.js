import React from 'react';
import { storiesOf } from '@storybook/react';

import Container from './index';

storiesOf('Container', module)
  .add('default', () => (
    <Container style={{ backgroundColor: 'pink' }}>
      <h1>Hello</h1>
    </Container>
  ));
