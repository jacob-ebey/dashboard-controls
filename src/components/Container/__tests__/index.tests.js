import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Container from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('components/Container - index.js', () => {
  const createProps = () => ({
    children: <h1>Hello, World</h1>,
  });

  test('Renders without error', () => {
    const props = createProps();

    const wrapper = shallow(<Container {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
