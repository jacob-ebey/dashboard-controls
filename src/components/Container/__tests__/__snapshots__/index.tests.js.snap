// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Container - index.js Renders without error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component>
    <h1>
      Hello, World
    </h1>
  </Component>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Hello, World
      </h1>,
      "className": "container ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "Hello, World",
      },
      "ref": null,
      "rendered": "Hello, World",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Hello, World
        </h1>,
        "className": "container ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Hello, World",
        },
        "ref": null,
        "rendered": "Hello, World",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
